lib/configure.js

```diff
@@ -137,6 +139,18 @@ function configure (gyp, argv, callback) {
     // disable -T "thin" static archives by default
     variables.standalone_static_library = gyp.opts.thin ? 0 : 1
 
+    if (win && !(gyp.opts.msvs_version && gyp.opts.msvs_version !== '2017')) {
```

Comment:
would you mind inverting this? reads really badly. `(win && (!gyp.opts.msvs_version || gyp.opts.msvs_version === '2017'))`